src/storage/src/region/writer.rs

```diff
@@ -139,6 +143,15 @@ impl RegionWriter {
             manifest_version,
             max_memtable_id,
         };
+        self.statistics_collector
+            .increase_disk_usage_bytes(version_edit.files_to_add.iter().map(|f| f.file_size).sum());
+        self.statistics_collector.descrease_disk_usage_bytes(
+            version_edit
+                .files_to_remove
+                .iter()
+                .map(|f| f.file_size)
+                .sum(),
+        );
```

Comment:
Since it isn't frequent to acquire the total file size, I think we could calculate the disk usage at `disk_usage_bytes()`. Then we don't need to introduce the statistic collector and maintain it.

RocksDB also uses a similar approach so it shouldn't be a bottleneck.
https://github.com/facebook/rocksdb/blob/50e9b3f9c7e9198257d0398bf14c8ddad70527cb/db/version_set.cc#L1779-L1783